algorithms/src/snark/marlin/ahp/matrices.rs

```diff
@@ -224,18 +206,36 @@ pub(crate) fn arithmetize_matrix<F: PrimeField>(
     MatrixArithmetization {
         row: LabeledPolynomial::new(labels.next().unwrap(), row, None, None),
         col: LabeledPolynomial::new(labels.next().unwrap(), col, None, None),
-        val: LabeledPolynomial::new(labels.next().unwrap(), val, None, None),
         row_col: LabeledPolynomial::new(labels.next().unwrap(), row_col, None, None),
+        row_col_val: LabeledPolynomial::new(labels.next().unwrap(), row_col_val, None, None),
         evals_on_K: matrix_evals,
     }
 }
 
+pub(crate) fn non_zero_entries_per_column<F: PrimeField>(
+    matrix: &Matrix<F>,
+    variable_domain: &EvaluationDomain<F>,
+    input_domain: &EvaluationDomain<F>,
+) -> Result<BTreeMap<usize, Vec<usize>>, anyhow::Error> {
+    let mut c_i_to_k = BTreeMap::<usize, Vec<usize>>::new();
+    let mut k = 0;
+    for row in matrix.iter() {
+        for (_, input_var_index) in row {
+            let c_i = variable_domain.reindex_by_subdomain(input_domain, *input_var_index)?;
+            c_i_to_k.entry(c_i).or_default().push(k);
+            k += 1;
+        }
+    }
+    Ok(c_i_to_k)
+}
```

Comment:
Can you add some documentation for what this function does? E.g. what is `k`?

Reply:
bfead31c1